frontend/cs/r1cs/api_assertions.go

```diff
@@ -128,16 +144,22 @@ func (builder *builder) mustBeLessOrEqVar(a, bound expr.LinearExpression) {
 		// note if bound[i] == 1, this constraint is (1 - ai) * ai == 0
 		// → this is a boolean constraint
 		// if bound[i] == 0, t must be 0 or 1, thus ai must be 0 or 1 too
-		builder.MarkBoolean(aBits[i].(expr.LinearExpression)) // this does not create a constraint
+		builder.MarkBoolean(aBits[i]) // this does not create a constraint
```

Comment:
Similarly to comment below, why do we mark `aBits[i]` boolean. It looks the output is not returned or used (except for the call to `builder.newR1C` below, but I think it doesn't check that input is boolean)

Reply:
I don't know why we do that, as you pointed out, it's not useful.
The only case where I could see this being useful, is for prover side optimizations. Since it "tags" the wire as being a boolean, we could in theory identify a lot of 0 and 1 in the vector of field elements at proving time to do specialized multi exps or ffts. 

Going to remove it for now, the day we add these kind of optim it's easy to restore.